api/v1alpha1/nodepool_types.go

```diff
@@ -419,11 +419,74 @@ type PowerVSNodePoolPlatform struct {
 	ImageDeletePolicy string `json:"imageDeletePolicy,omitempty"`
 }
 
+// KubevirtCompute contains values associated with the virtual compute hardware requested for the VM.
+type KubevirtCompute struct {
+	// Memory represents how much guest memory the VM should have
+	//
+	// Defaults to 4Gi
+	//
+	// +optional
+	// +kubebuilder:default="4Gi"
+	Memory *resource.Quantity `json:"memory,omitempty"`
+
+	// Memory represents how many cores the guest VM should have
+	//
+	// Defaults to 2 cores
+	//
+	// +optional
+	// +kubebuilder:default=2
+	Cores *uint32 `json:"cores,omitempty"`
+}
+
+// KubevirtPersistentVolume containes the values involved with provisioning persistent storage for a KubeVirt VM.
+type KubevirtPersistentVolume struct {
+	// Size is the size of the persistent storage volume
+	//
+	// +optional
+	// +kubebuilder:default="16Gi"
+	Size *resource.Quantity `json:"size,omitempty"`
+	// StorageClass is the storageClass used for the underlying PVC that hosts the volume
+	//
+	// +optional
+	StorageClass *string `json:"storageClass,omitempty"`
+}
+
+// KubevirtRootVolume represents the volume that the rhcos disk will be stored and run from.
+type KubevirtRootVolume struct {
+	// Image represents what rhcos image to use for the node pool
+	Image *KubevirtDiskImage `json:"diskImage,omitempty"`
```

Comment:
as a omitempty pointer this is optional, though it'd be good to add the +optional for consistency/readibility.